lib/yard_sorbet/struct_handler.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+# Handle all `const` calls, creating accessor methods, and compiles them for later usage at the class level
+# in creating a constructor
+class YARDSorbet::StructHandler < YARD::Handlers::Ruby::Base
+  handles method_call(:const)
+  namespace_only
+
+  def process
+    # Store the property for use in the constructor definition
+    name = statement.parameters[0].jump(:ident).source
+    doc = statement.docstring.to_s
+    source = statement.source
+    types = YARDSorbet::SigToYARD.convert(statement.parameters[1])
+    default_node = statement.traverse { |n| break n if n.source == 'default:' && n.type == :label }
+    default = default_node.parent[1].source if default_node
+
+    extra_state.prop_docs ||= Hash.new { |h, k| h[k] = [] }
+    extra_state.prop_docs[namespace] << {
+      doc: doc,
+      prop_name: name,
+      types: types,
+      source: source,
+      default: default
+    }
+
+    # Create the virtual method in our current scope
+    namespace.attributes[scope][name] ||= SymbolHash[read: nil, write: nil]
+
+    object = MethodObject.new(namespace, name, scope)
+    object.source = source
+
+    register(object)
+
+    reader_docstring = doc.empty? ? "Returns the value of attribute +#{name}+." : doc
+    docstring = YARD::DocstringParser.new.parse(reader_docstring).to_docstring
+    docstring.add_tag(YARD::Tags::Tag.new(:return, '', types))
+    object.docstring = docstring.to_raw
+
+    # Register the object explicitly as an attribute
+    namespace.attributes[scope][name][:read] = object
+  end
+end
+
+# Class-level handler that folds all `const` declarations into the constructor documentation
+# this needs to be injected as a module otherwise the default Class handler will overwrite documentation
+module YARDSorbet::StructClassHandler
+  def process
+    ret = super
+
+    return ret if extra_state.prop_docs.nil?
+
+    # lookup the full YARD path for the current class
+    class_ns = YARD::CodeObjects::ClassObject.new(namespace, statement[0].source.gsub(/\s/, ''))
+    return ret if extra_state.prop_docs[class_ns].empty?
+
+    props = extra_state.prop_docs[class_ns]
+
+    # Create a virtual `initialize` method with all the `prop`/`const` arguments
+    # having the name :initialize & the scope :instance marks this as the constructor
```

Comment:
What happens if the struct has an explicit `initialize` method? (I have seen this, for example to raise an `ArgumentError` if an `Integer` is outside an expected range.)

Reply:
hm, we haven't had a use for that yet (we validate args elsewhere). i can see how to make sure that the existing docstring is retained, if any. imo it would be an anti-pattern to add any docs there, really; you should add constraints (if any) on the prop itself